js/ui/check_box.js

```diff
@@ -81,6 +81,23 @@ var CheckBox = Editor.inherit({
         });
     },
 
+    _focusInHandler(e) {
```

Comment:
What do you think about use styles to resolve the initial issue?
```suggestion
  _focusInHandler(e) {
     if(this.option("validationMessageMode") === "auto") {
            if(this._$validationMessage) {
                this._$validationMessage.addClass("dx-hide-validation-message");
            }

            setTimeout(() => this._$validationMessage.removeClass("dx-hide-validation-message"), 100);
        }

         this.callBase(e);
    },
```
+ update our `validation.less`
+ move fix to the editor (?) - similar behavior is related to the RadioGroup